### PR TITLE
feat: add snacks to api reference docs

### PR DESCRIPTION
**Pull request by WoLewicki**
Added snacks to almost all examples and fixed some typos etc.

### CODE REVIEW

File: website/static/examples/next/drawer-actions.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  );
+}
+
+function CustomDrawerContent(props) {
+  return (
+    <DrawerContentScrollView {...props}>
+      <DrawerItemList {...props} />
+      <DrawerItem
+        label="Close drawer"
+        onPress={() => navRef.current.dispatch(DrawerActions.closeDrawer())}
```
Comment (osdnk) [inline website/static/examples/next/drawer-actions.js]: Maybe `useNavigation()` and then just `openDrawer()`

File: website/static/examples/next/drawer-dispatch.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import { NavigationNativeContainer } from '@react-navigation/native';
+import {
+  createDrawerNavigator,
+  DrawerContentScrollView,
+  DrawerItemList,
+  DrawerItem,
+} from '@react-navigation/drawer';
+import { DrawerActions } from '@react-navigation/routers';
+
+const navRef = React.createRef(null);
```
Comment (osdnk) [inline website/static/examples/next/drawer-dispatch.js]: Same

File: website/static/examples/next/drawer-open-close-toggle.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import { NavigationNativeContainer } from '@react-navigation/native';
+import {
+  createDrawerNavigator,
+  DrawerContentScrollView,
+  DrawerItemList,
+  DrawerItem,
+} from '@react-navigation/drawer';
+import { DrawerActions } from '@react-navigation/routers';
+
+const navRef = React.createRef(null);
```
Comment (osdnk) [inline website/static/examples/next/drawer-open-close-toggle.js]: Same

File: website/static/examples/next/simple-native-stack.js [hunk 4]
```diff
@@ -22,6 +22,7 @@ function Notifications({ navigation }) {
         title="Go to Profile"
         onPress={() => navigation.navigate('Profile')}
       />
+      <Button title="Go Back" onPress={() => navigation.goBack()} />
```
Comment (osdnk) [inline website/static/examples/next/simple-native-stack.js]: Just `navigation.goBack`

File: website/static/examples/next/simple-focus-and-blur.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    </View>
+  );
+}
+
+function ProfileScreen({ navigation }) {
+  React.useEffect(() =>
+    navigation.addListener('focus', () => alert('Screen was focused'))
+  );
+
+  React.useEffect(() =>
```
Comment (osdnk) [inline website/static/examples/next/simple-focus-and-blur.js]: Add array as a second arg (not only here)

File: website/static/examples/next/navigation-context.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        />
+      </View>
+    );
+  }
+}
+
+function ProfileScreen({ navigation }) {
+  return (
+    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+      <Button title="Go back" onPress={() => navigation.goBack()} />
```
Comment (osdnk) [inline website/static/examples/next/navigation-context.js]: Same

File: website/static/examples/next/native-stack-push-pop.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  return (
+    <Stack.Navigator>
+      <Stack.Screen name="Home" component={HomeScreen} />
+      <Stack.Screen name="Profile" component={ProfileScreen} />
+    </Stack.Navigator>
+  );
+}
+
+export default function App() {
+  enableScreens();
```
Comment (osdnk) [inline website/static/examples/next/native-stack-push-pop.js]: I think it’s more common to do it in global scope

File: docs/material-top-tab-navigator.md [hunk 8]
```diff
@@ -270,6 +274,7 @@ function MyTabBar({ state, descriptors, navigation, position }) {
           });
         };
 
+        const inputRange = [0, 1, 2];
```
Comment (satya164) [inline docs/material-top-tab-navigator.md]: To keep it in sync with number of tabs, I'd do

```suggestion
        const inputRange = state.routes.map((_, i) => i);
```

File: website/static/examples/next/drawer-actions.js (hunk 1 above)
Comment (satya164) [reply website/static/examples/next/drawer-actions.js]: You also get `navigation` in props

Comment (WoLewicki) [reply website/static/examples/next/drawer-actions.js]: I will use navigation from props then :)

**Pull request merged**